mex/common/models/extracted_data.py

```diff
@@ -35,23 +61,42 @@ def __str__(self) -> str:
 
 
 class ExtractedData(BaseExtractedData):
-    """Base model class for extracted data instances that ensures identities."""
+    """Base model class for extracted data items that ensures identities.
+
+    This base class does not add any attributes. It only adds the functionality
+    to automatically set identifiers for provenance. See below, for description.
+    """
 
     @root_validator(pre=True)
     def set_identifiers(cls, values: dict[str, Any]) -> dict[str, Any]:
         """Ensure identifier and provenance attributes are set for this instance.
 
-        A lookup is performed to determine whether this extracted data instance already
-        has an `identifier` or `stableTargetId`.
-        If not, new ones are generated and the association remembered.
-
-        If the `identifier` field has been set manually, e.g. passed to the constructor,
-        we check that it is already present in the identity provider and is assigned to
-        the same extracted data instance: it must be the same combination of
-        `identifier`, `hadPrimarySource` and `identifierInPrimarySource`.
-        If the identity is not found or the `identifier` differs, an error is thrown.
-        An exception is made for the MEx primary source which serves as the root node
-        for all relations: it may the `stableTargetId` manually.
+        All extracted data classes have four important identifiers that are defined
+        by `MExModel` and `BaseExtractedData`:
+
+        - identifierInPrimarySource
+        - hadPrimarySource
+        - identifier
+        - stableTargetID
+
+        Every time we create a new instance of an extracted item, we validate that these
+        identifiers are set correctly.
+
+        We check that `identifierInPrimarySource` and `hadPrimarySource` are provided,
+        because otherwise we cannot reliably determine the origin of this item.
+
+        Next we query the configured `IdentityProvider` to determine whether this item
+        already has an `identifier` and `stableTargetId`.
+        If not, we let the identity provider generate new identifiers.
+
+        If an `identifier` has been passed to the constructor, we check that it matches
+        with what we got from the identity provider, because we don't allow any system
+        to change the association from `identifierInPrimarySource` and
+        `hadPrimarySource` to `identifier`.
+
+        If a `stableTargetID` has been passed to the constructor, we use that as the
+        new value, because changes to the stable target ID are generally allowed.
+        For example, when matching to extracted items that describe the same thing.
```

Comment:
After reading this part I am confused again. What is exactly the purpose of the "stable" target ID, when changes are generally allowed?

Reply:
tried to address that here: https://github.com/robert-koch-institut/mex-common/pull/36/commits/6978ccd6f76231056b1d163d294379cb18b3d11c#diff-aa31d93942aa11a22383bc209495b8c58de08471a2e690d69a0a05770f5b6817R170